core/injector/application_injector.cpp

```diff
@@ -545,19 +539,27 @@ namespace {
 
     log->warn(
         "Can not obtain a libp2p keypair from crypto storage. "
-        "A unique one will be generated for the current session");
-
-    auto generated_keypair = crypto_provider.generateKeypair();
-
-    auto &&pub = generated_keypair.public_key;
-    auto &&priv = generated_keypair.secret_key;
+        "A unique one will be generated");
+
+    kagome::crypto::Ed25519Keypair generated_keypair;
+    auto save = app_config.saveNodeKey();
+    if (save) {
+      auto res = const_cast<crypto::CryptoStore &>(crypto_store)
```

Comment:
Er, I don't like the const_cast, perhaps we should pass it as a mutable reference in this case.

Reply:
Also, it would be great to move this logic elsewhere, since stuff like this in the injector code may eventually bloat this source file.